app/uk/gov/hmrc/emcstfereferencedata/repositories/ExciseProductsRepository.scala

```diff
@@ -81,4 +83,31 @@ class ExciseProductsRepository @Inject() (val mongoComponent: MongoComponent)(us
       .find()
       .sort(ascending("code"))
       .toFuture()
+
+  def fetchProductCodesInformation(
+    cnInformationRequest: CnInformationRequest
+  ): Future[Map[String, CnCodeInformation]] = {
+
+    val productCodeToCnCode: Map[String, String] =
+      cnInformationRequest.items.map(item => item.productCode -> item.cnCode).toMap
```

Comment:
Hmm just a thought but I think that if there are multiple CN codes and product codes that are the same, this will lose "duplicate" product <-> CN code mappings